File: tea-manage/src/app/shared/models/discount.ts
import { PaginationRequest } from './base';

export interface Discount {
  id: number;
  name: string;
  promotionCode: string;
  amountOff?: number;
  percentOff?: number;
}

export interface DiscountAdd {
  name: string;
  promotionCode: string;
  amountOff?: number;
  percentOff?: number;
}

export interface DiscountUpdate {
  id: number;
  name: string;
  promotionCode: string;
  amountOff?: number;
  percentOff?: number;
}

export class DiscountParams extends PaginationRequest {
  constructor() {
    super();
  }
}

File: tea-server/src/Tea.Domain/Entities/Discount.cs
namespace Tea.Domain.Entities
{
    public class Discount
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public decimal? AmountOff { get; set; }
        public decimal? PercentOff { get; set; }
        public required string PromotionCode { get; set; }
        public bool IsDeleted { get; set; }
    }
}
